src/ssh/cockpitsshrelay.c

```diff
@@ -1374,12 +1374,18 @@ cockpit_ssh_connect (CockpitSshData *data,
     }
 
   g_warn_if_fail (ssh_options_set (data->session, SSH_OPTIONS_HOST, host) == 0);
+  if (port != 0)
+    g_warn_if_fail (ssh_options_set (data->session, SSH_OPTIONS_PORT, &port) == 0);
 #if LIBSSH_085
     g_warn_if_fail (ssh_options_parse_config (data->session, NULL) == 0);
 #endif
-  g_warn_if_fail (ssh_options_set (data->session, SSH_OPTIONS_USER, data->username) == 0);
-  if (port != 0)
-    g_warn_if_fail (ssh_options_set (data->session, SSH_OPTIONS_PORT, &port) == 0);
+
+    /* If the ssh config doesn't specify User, use the username from the auth message */
+  gchar *username;
```

Comment:
You can use `g_autofree gchar *username = NULL` here and drop the explicit `g_free()`. We started using g_auto a while ago. Completely optional, of course.

Reply:
ooh, nice